xmtp_mls/src/storage/sql_key_store.rs

```diff
@@ -22,60 +31,951 @@ impl<'a> SqlKeyStore<'a> {
+        ProposalRef: traits::ProposalRef<CURRENT_VERSION>,
+        QueuedProposal: traits::QueuedProposal<CURRENT_VERSION>,
+    >(
+        &self,
+        group_id: &GroupId,
+        proposal_ref: &ProposalRef,
+        proposal: &QueuedProposal,
+    ) -> Result<(), Self::Error> {
+        // write proposal to key (group_id, proposal_ref)
+        let key = serde_json::to_vec(&(group_id, proposal_ref))?;
+        let value = serde_json::to_vec(proposal)?;
+        self.write::<CURRENT_VERSION>(QUEUED_PROPOSAL_LABEL, &key, &value)?;
+
+        // update proposal list for group_id
+        let key = serde_json::to_vec(group_id)?;
+        let value = serde_json::to_vec(proposal_ref)?;
+        self.append::<CURRENT_VERSION>(PROPOSAL_QUEUE_REFS_LABEL, &key, &value)?;
+
+        Ok(())
+    }
+
+    fn write_tree<
+        GroupId: traits::GroupId<CURRENT_VERSION>,
+        TreeSync: traits::TreeSync<CURRENT_VERSION>,
+    >(
+        &self,
+        group_id: &GroupId,
+        tree: &TreeSync,
+    ) -> Result<(), Self::Error> {
+        self.write::<CURRENT_VERSION>(
+            TREE_LABEL,
+            &serde_json::to_vec(&group_id).unwrap(),
+            &serde_json::to_vec(&tree).unwrap(),
+        )
+    }
+
+    fn write_interim_transcript_hash<
+        GroupId: traits::GroupId<CURRENT_VERSION>,
+        InterimTranscriptHash: traits::InterimTranscriptHash<CURRENT_VERSION>,
+    >(
+        &self,
+        group_id: &GroupId,
+        interim_transcript_hash: &InterimTranscriptHash,
+    ) -> Result<(), Self::Error> {
+        let mut values = self.values.write().unwrap();
+        let key = build_key::<CURRENT_VERSION, &GroupId>(INTERIM_TRANSCRIPT_HASH_LABEL, group_id);
+        let value = serde_json::to_vec(&interim_transcript_hash).unwrap();
+
+        values.insert(key, value);
+        Ok(())
+    }
+
+    fn write_context<
+        GroupId: traits::GroupId<CURRENT_VERSION>,
+        GroupContext: traits::GroupContext<CURRENT_VERSION>,
+    >(
+        &self,
+        group_id: &GroupId,
+        group_context: &GroupContext,
+    ) -> Result<(), Self::Error> {
+        let mut values = self.values.write().unwrap();
+        let key = build_key::<CURRENT_VERSION, &GroupId>(GROUP_CONTEXT_LABEL, group_id);
+        let value = serde_json::to_vec(&group_context).unwrap();
+
+        values.insert(key, value);
+        Ok(())
+    }
+
+    fn write_confirmation_tag<
+        GroupId: traits::GroupId<CURRENT_VERSION>,
+        ConfirmationTag: traits::ConfirmationTag<CURRENT_VERSION>,
+    >(
+        &self,
+        group_id: &GroupId,
+        confirmation_tag: &ConfirmationTag,
+    ) -> Result<(), Self::Error> {
+        let mut values = self.values.write().unwrap();
+        let key = build_key::<CURRENT_VERSION, &GroupId>(CONFIRMATION_TAG_LABEL, group_id);
+        let value = serde_json::to_vec(&confirmation_tag).unwrap();
+
+        values.insert(key, value);
+        Ok(())
+    }
+
+    fn write_signature_key_pair<
```

Comment:
I'm not sure if you're actually using the signature key functions. You may want to skip the unused functions to make sure they're not used.

Reply:
Return not implemented errors.

Reply:
That's an example, right? I'd imagine we are using the `SignatureKeyPair` methods, since that one seems pretty important.